astropy/table/soco.py

```diff
@@ -0,0 +1,170 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+The SCEngine class uses the `sortedcontainers` package to implement an
+Index engine for Tables.
+"""
+
+from collections import OrderedDict
+from itertools import starmap
+
+try:
+    from sortedcontainers import SortedList
+    HAS_SOCO = True
+except ImportError:
+    HAS_SOCO = False
+
+
+class Node(object):
+    __slots__ = ('key', 'value')
+
+    def __init__(self, key, value):
+        self.key = key
+        self.value = value
+
+    def __lt__(self, other):
+        if other.__class__ is Node:
+            return (self.key, self.value) < (other.key, other.value)
+        return self.key < other
+
+    def __le__(self, other):
+        if other.__class__ is Node:
+            return (self.key, self.value) <= (other.key, other.value)
+        return self.key <= other
+
+    def __eq__(self, other):
+        if other.__class__ is Node:
+            return (self.key, self.value) == (other.key, other.value)
+        return self.key == other
+
+    def __ne__(self, other):
+        if other.__class__ is Node:
+            return (self.key, self.value) != (other.key, other.value)
+        return self.key != other
+
+    def __gt__(self, other):
+        if other.__class__ is Node:
+            return (self.key, self.value) > (other.key, other.value)
+        return self.key > other
+
+    def __ge__(self, other):
+        if other.__class__ is Node:
+            return (self.key, self.value) >= (other.key, other.value)
+        return self.key >= other
+
+    __hash__ = None
+
+    def __repr__(self):
+        return 'Node({0!r}, {1!r})'.format(self.key, self.value)
+
+
+class SCEngine:
+    '''
+    Fast tree-based implementation for indexing, using the
+    `sortedcontainers` package.
```

Comment:
Could you use double backticks otherwise sphinx fails when tries to create a link for this.